projects/aca-content/src/lib/components/search/search-input/search-input.component.spec.ts

```diff
@@ -239,4 +239,205 @@ describe('SearchInputComponent', () => {
+    });
+
+    it('should sync input values and reset has400LibraryError', () => {
+      component.has400LibraryError = true;
+
+      component.searchByOption();
+
+      expect(syncInputValuesSpy).toHaveBeenCalled();
+      expect(component.has400LibraryError).toBeFalse();
+    });
+
+    it('should evaluate libraries constraint when libraries are checked', () => {
+      isLibrariesCheckedSpy.and.returnValue(true);
+
+      component.searchByOption();
+
+      expect(evaluateLibrariesConstraintSpy).toHaveBeenCalled();
+      expect(component.hasLibrariesConstraint).toBeFalse();
+    });
+
+    it('should set hasLibrariesConstraint based on evaluateLibrariesConstraint result', () => {
+      isLibrariesCheckedSpy.and.returnValue(true);
+      evaluateLibrariesConstraintSpy.and.returnValue(true);
+
+      component.searchByOption();
+
+      expect(component.hasLibrariesConstraint).toBeTrue();
+    });
+
+    it('should update queryLibrariesBuilder when libraries are checked, onLibrariesSearchResults is true, and search term changed', () => {
+      isLibrariesCheckedSpy.and.returnValue(true);
+      spyOnProperty(component, 'onLibrariesSearchResults', 'get').and.returnValue(true);
+      isSameSearchTermSpy.and.returnValue(false);
+
+      component.searchedWord = 'test';
+
+      component.searchByOption();
+
+      expect(queryLibrariesBuilderUpdateSpy).toHaveBeenCalled();
+      expect(storeDispatchSpy).not.toHaveBeenCalled();
+    });
+
+    it('should dispatch SearchByTermAction when libraries are checked, onLibrariesSearchResults is false, and searchedWord is set', () => {
+      isLibrariesCheckedSpy.and.returnValue(true);
+      spyOnProperty(component, 'onLibrariesSearchResults', 'get').and.returnValue(false);
+
+      component.searchedWord = 'test';
+
+      component.searchByOption();
+
+      expect(storeDispatchSpy).toHaveBeenCalledWith(new SearchByTermAction('test', component.searchOptions));
+      expect(queryLibrariesBuilderUpdateSpy).not.toHaveBeenCalled();
+    });
+
+    it('should dispatch SearchByTermAction when libraries are checked, onLibrariesSearchResults is true, search term same, and searchedWord is set', () => {
+      isLibrariesCheckedSpy.and.returnValue(true);
+      spyOnProperty(component, 'onLibrariesSearchResults', 'get').and.returnValue(true);
+      isSameSearchTermSpy.and.returnValue(true);
+
+      component.searchedWord = 'test';
+
+      component.searchByOption();
+
+      expect(storeDispatchSpy).toHaveBeenCalledWith(new SearchByTermAction('test', component.searchOptions));
+      expect(queryLibrariesBuilderUpdateSpy).not.toHaveBeenCalled();
+    });
+
+    it('should add folder filter when only folders are checked', () => {
+      isLibrariesCheckedSpy.and.returnValue(false);
+      isFoldersCheckedSpy.and.returnValue(true);
+      isFilesCheckedSpy.and.returnValue(false);
+
+      spyOn(component['queryBuilder'], 'addFilterQuery').and.callThrough();
```

Comment:
it's a bit hacking to get private field using ['queryBuilder']. Private field is private so should not be accessible. Instead you should get this field using injection (please use TestBed.inject(SearchQueryBuilderService). 

The same issue is in other tests so please correct them as well.